### PR TITLE
Add code used to find dependency tree of a specific package

### PR DESCRIPTION
**Pull request by ryzngard**
image

### CODE REVIEW

**Comment by davidwengier** [comment]
ooohhh this is interesting... I've been doing this manually with VS Code's find in files, looking through `.nuspec` files in my nuget cache, so this could be _very_ handy!

File: src/ProjectDependencies/BuildDependencyFinder.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+using System.Threading.Tasks;
+using NuGet.Packaging.Core;
+
+namespace ProjectDependencies
+{
+    public static class BuildDependencyFinder
+    {
+        static readonly LockFileFormat s_lockFileFormat = new();
+
+        public static async Task<ImmutableArray<DependencyNode>> FindDependenciesAsync(string folder, string packageName, string packageVersion, Action<int, bool> callback, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (davidwengier) [reply src/ProjectDependencies/BuildDependencyFinder.cs:23]: Where else would i keep my `project.asset.json` files, if not on my phone?

[the remark this answers is not in the document]
Comment (ryzngard) [reply src/ProjectDependencies/BuildDependencyFinder.cs:23]: the future dev machine was in your pocket the whole time

File: src/ProjectDependencies/App.xaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿<?xml version = "1.0" encoding = "UTF-8" ?>
```
Comment (dibarbet) [inline src/ProjectDependencies/App.xaml:1]: i will not pretend to understand any of the maui bits, but dang thats a lot of boilerplate 😆

File: src/ProjectDependencies/ProjectDependencies.csproj [hunk 3]
```diff
@@ -0,0 +1,73 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+	<Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
+
+	<PropertyGroup>
+		<TargetFrameworks>net7.0-maccatalyst</TargetFrameworks>
+		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);net7.0-windows10.0.19041.0</TargetFrameworks>
+		<!-- Uncomment to also build the tizen app. You will need to install tizen by following this: https://github.com/Samsung/Tizen.NET -->
```
Comment (dibarbet) [inline src/ProjectDependencies/ProjectDependencies.csproj:8]: needed?

**Review by dibarbet — APPROVED** [review]

File: src/ProjectDependencies/ProjectDependencies.csproj (hunk 3 above)
Comment (ryzngard) [reply src/ProjectDependencies/ProjectDependencies.csproj:8]: probably not 🤷

This is all set up by MAUI for me

**Pull request merged**